quint/mempool/Consensus.qnt

```diff
@@ -0,0 +1,180 @@
+// -*- mode: Bluespec; -*-
+// The common consensus abstraction where processes agree on some blocks (i.e., set of transactions).
+
+module Consensus{
+
+    import Spells.* from "Spells"
+    import System.* from "System"
+
+    type ConsensusState = {
+      proposed: Process -> Set[Tx],
+      decided: Process -> Set[Tx]
+    }
+
+    pure def newConsensus(P: Set[Process]): ConsensusState = {
+      proposed: P.mapBy(p => Set()),
+      decided: P.mapBy(p => Set())
+    }
+
+    pure def processes(state: ConsensusState): Set[Process] = {
```

Comment:
If you like one-liners, most of these definitions can be written without the brackets 😉